crates/debug/src/transform/expression.rs

```diff
@@ -77,47 +112,41 @@ fn translate_loc(
     isa: &dyn TargetIsa,
     add_stack_value: bool,
 ) -> Result<Option<Vec<u8>>> {
-    use gimli::write::Writer;
     Ok(match loc {
         ValueLoc::Reg(reg) if add_stack_value => {
-            let machine_reg = isa.map_dwarf_register(reg)? as u8;
-            Some(if machine_reg < 32 {
-                vec![gimli::constants::DW_OP_reg0.0 + machine_reg]
+            let machine_reg = isa.map_dwarf_register(reg)?;
+            let mut writer = GimliWriter::new();
+            if machine_reg < 32 {
+                writer.write_op_reg(gimli::constants::DW_OP_reg0, machine_reg)?;
```

Comment:
Why not move this `if machine_reg < 32` logic into `GimliWriter::write_op_reg`? That could remove a precondition assertion from that method and some duplication here and below.